server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/InfinispanSubsystemXMLReader.java

```diff
@@ -343,6 +343,21 @@ private void parseContainer(XMLExtendedStreamReader reader, PathAddress subsyste
                 }
             }
         }
+
+        addSyntheticHealtheOperation(operations, containerAddress);
+    }
+
+    /**
+     * A {@link HealthResource} is not populated to the XML. However the easiest way to propagate a Health node
+     * (we need to attach Health node to the Cache Container node) and attach metrics to it is to
+     * simply simulate reading it.
+     *
+     * Since we are not persisting it - there is no need to check the namespaces.
+     */
+    private void addSyntheticHealtheOperation(Map<PathAddress, ModelNode> operations, PathAddress containerAddress) {
+        PathAddress healthAddress = containerAddress.append(ModelKeys.HEALTH, ModelKeys.HEALTH_NAME);
```

Comment:
I think this approach is wrong. health is a runtime resource and therefore it should not be handled by the configuration model